doc/reference/javascript_cheatsheet.rst

```diff
@@ -276,4 +276,236 @@ using the view in the kanban arch (a specific example is the helpdesk dashboard)
+
+        .. code-block:: javascript
+
+            function returnAsSoonAsAsyncProcessIsDone() {
+                var prom = AsyncProcess();
+                prom.then(function (resultOfAsyncProcess) {
+                        return doSomething();
+                });
+                /* returns prom which will only wait for AsyncProcess(),
+                   and when it will be resolved, the result will be the one of AsyncProcess */
+                return prom;
+            }
+
+Error handling
+--------------
+- in general in promises
+        The general idea is that a promise should not be rejected for control flow, but should only be rejected for errors.
+        When that is the case, you would have multiple resolutions of your promise with, for instance status codes that you would have to check in the `then` handlers and a single `catch` handler at the end of the promise chain.
+
+        .. code-block:: javascript
+
+            function a() {
+                x.y();  // <-- this is an error: x is undefined
+                return Promise.resolve(1);
+            }
+            function b() {
+               return Promise.reject(2);
+            }
+
+            a().catch(console.log);           // will log the error in a
+            a().then(b).catch(console.log);   // will log the error in a, the then is not executed
+            b().catch(console.log);           // will log the rejected reason of b (2)
+            Promise.resolve(1)
+                   .then(b)                   // the then is executed, it executes b
+                   .then(...)                 // this then is not executed
+                   .catch(console.log);       // will log the rejected reason of b (2)
+
+
+
+- in Odoo specifically
+        In Odoo, it happens that we use promise rejection for control flow, like in mutexes and other concurrency primitives defined in module `web.concurrency`
+        We also want to execute the catch for *business* reasons, but not when there is a coding error in the definition of the promise or of the handlers.
+        For this, we have introduced the concept of `guardedCatch`. It is called like `catch` but not when the rejected reason is an error
+
+        .. code-block:: javascript
+
+                var promiseWithError = Promise.resolve().then(function () {
+                    x.y();  // <-- this is an error: x is undefined
+                });
+                promiseWithError.guardedCatch(function (reason) {console.log(reason);}); // will not be called
+                promiseWithError.catch(function (reason) {console.log(reason);}); // will be called
+
+
+
+Testing asynchronous code
+--------------------------
+
+- using promises in tests
+        In the tests code, we support the latest version of Javascript, including primitives like `async` and `await`. This makes using and waiting for promises very easy.
```

Comment:
I think it would be useful to warn here that only ES8-compatible browsers will run the tests because of this.